app/src/main/java/com/testtask/apptesttask/model/repository/CharactersRepository.kt

```diff
@@ -14,31 +14,55 @@ class CharactersRepository @Inject constructor(
     private val marvelService: MarvelService,
     private val appConfig: AppConfig,
     private val schedulers: SchedulersProvider,
-    private val sharedPreference: Prefs
+    private val prefs: Prefs
 ) {
 
-    private fun getApiCharacters(): Single<List<ApiCharacter>> = marvelService
+    private fun getApiCharacters() = marvelService
             .getCharacters(appConfig.ts, appConfig.publicKey, appConfig.hash)
-            .flatMap { characterDataWrapper -> Single.just(characterDataWrapper.data.results) }
+            .flatMap { characterDataWrapper ->
+                Single.just(characterDataWrapper.data.results)
+            }
 
-    fun getCharacters(): Single<List<Character>> = Single
-            .zip(
+    private fun getPrefs(): Single<Map<Int, Character>> =
```

Comment:
Remove this method and use `Single.just(prefs.favoritesCharacters)` in `Single.zip`.